code/modules/mob/living/carbon/xenomorph/castes/defiler/abilities_defiler.dm

```diff
@@ -1,43 +1,110 @@
-	C.apply_damage(100, STAMINA)
-	C.apply_damage(10, BRUTE, "chest", updating_health = TRUE)
-	C.emote("scream")
+	X.face_atom(living_target)
+	X.do_attack_animation(living_target)
+	playsound(living_target, 'sound/effects/spray3.ogg', 15, TRUE)
+	playsound(living_target, pick('sound/voice/alien_drool1.ogg', 'sound/voice/alien_drool2.ogg'), 15, 1)
+	to_chat(X, "<span class='xenodanger'>Our stinger successfully discharges accelerant into our victim.</span>")
+	to_chat(living_target, "<span class='danger'>You feel horrible pain as something sharp forcibly pierces your thorax.</span>")
+	living_target.apply_damage(50, STAMINA)
+	living_target.apply_damage(5, BRUTE, "chest", updating_health = TRUE)
+	living_target.emote("scream")
+	var/defile_strength_multiplier = 1
+
+	var/hemodile_amount = living_target.reagents.get_reagent_amount(/datum/reagent/toxin/xeno_hemodile)
+	var/transvitox_amount = living_target.reagents.get_reagent_amount(/datum/reagent/toxin/xeno_transvitox)
+	var/neurotoxin_amount = living_target.reagents.get_reagent_amount(/datum/reagent/toxin/xeno_neurotoxin) + living_target.reagents.get_reagent_amount(/datum/reagent/toxin/xeno_neurotoxin/light)
+	if(living_target.reagents.get_reagent_amount(/datum/reagent/toxin/xeno_hemodile)) //Each other Defiler toxin increases the multiplier by 2x; 2x if we have 2 combo chem, 4x if we have all 3
+		defile_strength_multiplier *= 2
+
+	if(living_target.reagents.get_reagent_amount(/datum/reagent/toxin/xeno_transvitox)) //Each other Defiler toxin increases the multiplier by 2x; 2x if we have 2 combo chem, 4x if we have all 3
+		defile_strength_multiplier *= 2
+
+	if(living_target.reagents.get_reagent_amount(/datum/reagent/toxin/xeno_neurotoxin))
+		defile_strength_multiplier *= 2
+
+	var/defile_power = (hemodile_amount + transvitox_amount + neurotoxin_amount) * defile_strength_multiplier //Total amount of toxin damage we deal
+
+	living_target.adjustToxLoss(defile_power) //Apply the toxin damage
+
+	var/purge_list = list(/datum/reagent/toxin/xeno_hemodile, /datum/reagent/toxin/xeno_transvitox, /datum/reagent/toxin/xeno_neurotoxin)
+
+	if(!LAZYLEN(purge_list))
+		return
+	var/count = LAZYLEN(purge_list)
+	for(var/datum/reagent/toxin/xeno_toxin as() in living_target.reagents.reagent_list) //Purge all the xeno chems
+		if(count < 1)
+			break
+		if(is_type_in_typecache(xeno_toxin, GLOB.defile_purge_list))
+			count--
+			living_target.reagents.remove_reagent(xeno_toxin.type,defile_power)
+
+	var/datum/effect_system/smoke_spread/xeno/sanguinal/blood_smoke = new(X)
+	var/turf/target_turf = get_turf(living_target)
+	message_admins("Defile Power: [defile_power] Blood Smoke Strength/Size (round): [round(clamp(defile_power*DEFILER_SANGUINAL_SMOKE_MULTIPLIER,1,3))]  Blood Smoke Strength/Size (ceiling): [CEILING(clamp(defile_power*DEFILER_SANGUINAL_SMOKE_MULTIPLIER,1,3),1)]")
+	//blood_smoke.strength = CEILING(clamp(defile_power*DEFILER_SANGUINAL_SMOKE_MULTIPLIER,1,3),1)
```

Comment:
looks like testing leftovers, delete this?
```suggestion
```

Reply:
Yeah I'm pushing an update today after work